staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -476,7 +476,18 @@ type DeploymentStatus struct {
 	// +optional
 	UnavailableReplicas int32 `json:"unavailableReplicas,omitempty" protobuf:"varint,5,opt,name=unavailableReplicas"`
 
-	// Represents the latest available observations of a deployment's current state.
+	// Represents the latest observations of a deployment's current state (described by the observedGeneration).
+	// Deployments are given the following conditions automatically:
+	// Available is true when the number of available replicas under the deployment's replica sets is
+	// greater than the user's defined minimum availability. The minimum availability is controlled by the
+	// update strategy - the Recreate strategy has a minimum availability of zero (always Available), and
+	// the Rolling strategy defaults to a minimum availability of zero unless replicas is greater than 1,
+	// the maxSurge parameter is specified, or maxUnavailable is smaller than the number of replicas.
+	// Progressing is true when the deployment is trying to reach minimum availability on a new replica set.
```

Comment:
it is trying to reach the full availability

Reply:
and can become false when the progress times out

Reply:
Tweaked this even more based on your comment.